File: JavaScript/6js 1/dashatize.js
/*
|--------------------------------------------------------------------------
| Dashatize it
|--------------------------------------------------------------------------
|
*/

function dashatize(num) {
   const n = String(Math.abs(num))
   let res = '';
   for (let i = 0; i < n.length; i++) {
      res += Number(n[i]) % 2 == 0 ? n[i] : `-${n[i]}-`;
   }
   const result = res.split('-').filter(i => i != '').join('-')
   return result == 'N-a-N' ? 'NaN' : result
}


function dashatize(num) {
   return isNaN(num) ? 'NaN' : num.toString().match(/([13579]|[02468]+)/g).join('-');
}


function dashatize(num) {
   return String(num)
      .replace(/([13579])/g, "-$1-")
      .replace(/--+/g, "-")
      .replace(/(^-|-$)/g, "")
}


function dashatize(num) {
   return num = num.toString().split("").map(function (c) {
      return c % 2 ? "-" + c + "-" : c;
   }).join("").split("-").filter(a => a != "").join("-");

}


console.log(dashatize(274), "2-7-4");
console.log(dashatize(5311), "5-3-1-1");
console.log(dashatize(86320), "86-3-20");
console.log(dashatize(974302), "9-7-4-3-02");
console.log(dashatize(NaN), "NaN");
console.log(dashatize(0), "0");
console.log(dashatize(-1), "1");
console.log(dashatize(-28369), "28-3-6-9");

File: Python/6py 1/dashatize.py
"""
|--------------------------------------------------------------------------
| Dashatize it
|--------------------------------------------------------------------------
|
"""

def dashatize(n):
   if n == None:
      return 'None'
   else:
      num = str(abs(n))
      res = ''
      for i in range(len(num)):
         res += num[i] if int(num[i]) % 2 == 0 else f'-{num[i]}-'
      return '-'.join(list(filter(lambda i: i != '' , res.split('-'))))


def dashatize(num):
   try:
      return ''.join(['-'+i+'-' if int(i)%2 else i for i in str(abs(num))]).replace('--','-').strip('-')
   except:
      return 'None'


def dashatize(num):
   num_str = str(num)
   for i in ['1','3','5','7','9']:
      num_str = num_str.replace(i,'-' + i + '-')
   return num_str.strip('-').replace('--','-')


from re import findall
def dashatize(num):
   return '-'.join(findall('([13579]|[02468]+)', str(num))) or 'None'


print(dashatize(274), "2-7-4")
print(dashatize(5311), "5-3-1-1")
print(dashatize(86320), "86-3-20")
print(dashatize(974302), "9-7-4-3-02")
print(dashatize(None), "None")